From 094e856b48c4316ed079c36b288e0e39706b9018 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Thu, 18 Dec 2014 07:54:25 +0800
Subject: [PATCH 128/441] before unregister the device check if the device has
 registered

Change-Id: Ida983cf53efdd3b4fa0d4538472be1db830d0a74
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-14082
Signed-off-by: Yang, Dong <dong.yang@intel.com>
Signed-off-by: Fabien Marotte <fabien.marotte@intel.com>
(cherry picked from commit c0c4d9dab83ef2230a80697dbde485f1c063ffca)
---
 drivers/misc/heci/heci-api.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/misc/heci/heci-api.c
+++ b/drivers/misc/heci/heci-api.c
@@ -229,7 +229,7 @@ dev->print_log(dev, "%s(): there is some
 
 dev->print_log(dev, "%s(): before wait_event_interruptible\n", __func__);
 	if (wait_event_interruptible(cl->rx_wait,
-			(dev->dev_state == HECI_DEV_ENABLED && 
+			(dev->dev_state == HECI_DEV_ENABLED &&
 			(cl->read_rb || HECI_CL_INITIALIZING == cl->state || HECI_CL_DISCONNECTED == cl->state || HECI_CL_DISCONNECTING == cl->state)))) {
 		printk(KERN_ALERT "%s(): woke up not in success; sig. pending = %d signal = %08lX\n", __func__, signal_pending(current), current->pending.signal.sig[0]);
 		return	-ERESTARTSYS;
@@ -630,6 +630,9 @@ EXPORT_SYMBOL_GPL(heci_register);
 
 void heci_deregister(struct heci_device *dev)
 {
+	if (heci_misc_device.parent == NULL)
+		return;
+
 	heci_dbgfs_deregister(dev);
 	misc_deregister(&heci_misc_device);
 	heci_misc_device.parent = NULL;
